Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,122 @@ extension NavigationMapView {
+                case .int(let intId): return Double(intId)
+                }
+            }
+        }
+        return nil
+    }
+
+    private func buildingIdentifier(at coordinate: CLLocationCoordinate2D, in3D extrudesBuildings: Bool = true) {
+        let screenCoordinate = mapView.point(for: coordinate, in: self)
+        if let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) {
+            mapView.visibleFeatures(at: screenCoordinate,
+                                    styleLayers: Set(identifiers),
+                                    completion: { [weak self] result in
+                                        guard let validSelf = self else { return }
+                                        if case .success(let features) = result {
+                                            if let identifier = self?.featureIdentifierToDouble(features: features) {
+                                                self?.foundBuildingIds.insert(identifier)
+                                            }
+                                        }
+                                    })
+        }
+    }
+
+    private func addBuildingsLayer(with identifiers: Set<Double>, in3D: Bool = false, extrudeAll: Bool = false) {
+        mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
+        if identifiers.isEmpty { return }
+        var highlightedBuildingsLayer = FillExtrusionLayer(id: IdentifierString.buildingExtrusionLayer)
+        highlightedBuildingsLayer.source = "composite"
+        highlightedBuildingsLayer.sourceLayer = "building"
+
+        if extrudeAll {
+            highlightedBuildingsLayer.filter = Exp(.eq) {
+                Exp(.get) {
+                    "extrude"
+                }
+                "true"
+            }
+        } else {
+            highlightedBuildingsLayer.filter = Exp(.all) {
+                Exp(.eq) {
+                    Exp(.get) {
+                        "extrude"
+                    }
+                    "true"
+                }
+                Exp(.inExpression) {
+                    Exp(.id)
+                    Exp(.literal) {
+                        identifiers.map({$0})
+                    }
+                }
+            }
+        }
+
+        if in3D {
+            highlightedBuildingsLayer.paint?.fillExtrusionHeight = .expression(
+                Exp(.interpolate) {
+                    Exp(.linear)
+                    Exp(.zoom)
+                    13
+                    0
+                    13.25
+                    Exp(.get) {
+                        "height"
+                    }
+                }
+            )
+        } else {
+            highlightedBuildingsLayer.paint?.fillExtrusionHeight = .constant(0.0)
+        }
+
+        highlightedBuildingsLayer.paint?.fillExtrusionBase = .expression(
+            Exp(.interpolate) {
```

Comment:
Can we also re-use this expression syntax (I can see only height/min_height difference)?

Reply:
Thanks for the suggestion! I'll extract these expressions as a local variable.